ios/BrightcovePlayer.m

```diff
@@ -129,20 +143,110 @@ - (void)playbackController:(id<BCOVPlaybackController>)controller playbackSessio
         if (self.onReady) {
             self.onReady(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackReady")
+                                 });
+        }
     } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventPlay) {
         _playing = true;
         if (self.onPlay) {
             self.onPlay(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackPlay")
+                                 });
+        }
     } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventPause) {
         _playing = false;
         if (self.onPause) {
             self.onPause(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackPause")
+                                 });
+        }
     } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventEnd) {
         if (self.onEnd) {
             self.onEnd(@{});
         }
+        if (self.onStatusEvent) {
```

Comment:
Why do we need a separate one for the onEnd, as we already have the entirely separate event callback for it?

Reply:
I saw three options:
* Create individual callbacks for each status -> pain for clients and doesn't scale
* Create one callback for all the 'new' status codes, but require clients to know that there are four status codes not included and subscribe to them and handle them separately. (No benefit to the client consuming this.(
* Create one new callback for all the 'new' status codes AND put this callback into the existing callbacks.  This is somewhat ugly, but it maintains backwards compatibility and I thought it was the cleanest interface for clients to consume.  

I'm happy to hear a better way.  It seems like we have to compromise no matter which solution we pick.